hypershift-operator/controllers/nodepool/haproxy.go

```diff
@@ -215,7 +217,19 @@ func apiServerProxyConfig(haProxyImage, cpoImage, externalAPIAddress, internalAP
 			},
 		},
 	}
-	if proxyAddr == "" {
+
+	// Check if no proxy contains the internal API address or kubernetes SVC, if so, use HAProxy.
+	containsSubstring := func(s string, substrings ...string) bool {
+		for _, substr := range substrings {
+			if strings.Contains(s, substr) {
+				return true
+			}
```

Comment:
The [apiServerAddr](https://github.com/openshift/hypershift/blob/8bf0e34259ee150a0e7183fe7d41fbe492d27ec2/kubernetes-default-proxy/kubernetes_default_proxy.go#L33) can be an IP and `spec.configuration.proxy.noProxy` ([ProxySpec](https://github.com/openshift/api/blob/62b890e4c7cd2970e3340894b5ffff04bc67b17c/config/v1/types_proxy.go#L46-L49)) could be a CIDR/IP so that may need to be checked as well.